ChangePolicyforIndexIT testChangePolicyForIndex test failure

Expected `complete`, but was `allocate`

**Build scan**:
https://gradle-enterprise.elastic.co/s/zmpvqpicbtda4
**Repro line**:
```
./gradlew ':x-pack:plugin:ilm:qa:multi-node:javaRestTest' --tests "org.elasticsearch.xpack.ilm.ChangePolicyforIndexIT.testChangePolicyForIndex" -Dtests.seed=EECC349DA8CBF60F -Dtests.security.manager=true -Dtests.locale=et -Dtests.timezone=Pacific/Majuro -Druntime.java=8
```

**Reproduces locally?**:
No, (but I don't have windows)
**Applicable branches**:
7.11
**Failure excerpt**:
```
org.elasticsearch.xpack.ilm.ChangePolicyforIndexIT > testChangePolicyForIndex FAILED |  
-- | --
  | org.junit.ComparisonFailure: expected:<[comple]te> but was:<[alloca]te> |  
  | at __randomizedtesting.SeedInfo.seed([EECC349DA8CBF60F:FDD60DC658E25FF]:0) |  
  | at org.junit.Assert.assertEquals(Assert.java:115) |  
  | at org.junit.Assert.assertEquals(Assert.java:144) |  
  | at org.elasticsearch.xpack.ilm.ChangePolicyforIndexIT.assertStep(ChangePolicyforIndexIT.java:132) |  
  | at org.elasticsearch.xpack.ilm.ChangePolicyforIndexIT.lambda$testChangePolicyForIndex$2(ChangePolicyforIndexIT.java:116) |  
  | at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1016) |  
  | at org.elasticsearch.xpack.ilm.ChangePolicyforIndexIT.testChangePolicyForIndex(ChangePolicyforIndexIT.java:116)
```

There were other test failures in the same build. Another was for a separate integration class (same suite) and failed similarly (allocated while expecting completed).
```
org.elasticsearch.xpack.ilm.TimeSeriesLifecycleActionsIT > testAllocateOnlyAllocation FAILED |  
-- | --
  | java.lang.AssertionError: |  
  | Expected: <{"phase":"cold","action":"complete","name":"complete"}> |  
  | but: was <{"phase":"cold","action":"allocate","name":"check-allocation"}> |  
  | at __randomizedtesting.SeedInfo.seed([EECC349DA8CBF60F:A87251BFB4215803]:0) |  
  | at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18) |  
  | at org.junit.Assert.assertThat(Assert.java:956) |  
  | at org.junit.Assert.assertThat(Assert.java:923) |  
  | at org.elasticsearch.xpack.ilm.TimeSeriesLifecycleActionsIT.lambda$testAllocateOnlyAllocation$20(TimeSeriesLifecycleActionsIT.java:281) |  
  | at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1016) |  
  | at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:989)
```

--- a/x-pack/plugin/ilm/qa/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/ilm/ChangePolicyforIndexIT.java
+++ b/x-pack/plugin/ilm/qa/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/ilm/ChangePolicyforIndexIT.java
@@ -64,8 +64,10 @@ public class ChangePolicyforIndexIT extends ESRestTestCase {
         Map<String, Phase> phases2 = new HashMap<>();
         phases2.put("hot", new Phase("hot", TimeValue.ZERO, singletonMap(RolloverAction.NAME, new RolloverAction(null, null, 1000L))));
         phases2.put("warm", new Phase("warm", TimeValue.ZERO,
-                singletonMap(AllocateAction.NAME, new AllocateAction(1, singletonMap("_name", "javaRestTest-1,javaRestTest-2"), null, null))));
-        LifecyclePolicy lifecyclePolicy2 = new LifecyclePolicy("policy_1", phases2);
+                singletonMap(AllocateAction.NAME,
+                    new AllocateAction(1, singletonMap("_name", "javaRestTest-0,javaRestTest-1,javaRestTest-2,javaRestTest-3"),
+                        null, null))));
+        LifecyclePolicy lifecyclePolicy2 = new LifecyclePolicy("policy_2", phases2);
         // PUT policy_1 and policy_2
         XContentBuilder builder1 = jsonBuilder();
         lifecyclePolicy1.toXContent(builder1, null);
@@ -118,7 +120,7 @@ public class ChangePolicyforIndexIT extends ESRestTestCase {
         // Check index is allocated on javaRestTest-1 and javaRestTest-2 as per policy_2
         Map<String, Object> indexSettings = getIndexSettingsAsMap(indexName);
         String includesAllocation = (String) indexSettings.get("index.routing.allocation.include._name");
-        assertEquals("javaRestTest-1,javaRestTest-2", includesAllocation);
+        assertEquals("javaRestTest-0,javaRestTest-1,javaRestTest-2,javaRestTest-3", includesAllocation);
     }
 
     private void assertStep(String indexName, StepKey expectedStep) throws IOException {

